lib/Service/Gateway/Signal/Gateway.php

```diff
@@ -62,21 +62,41 @@ public function __construct(IClientService $clientService,
 	 */
 	public function send(IUser $user, string $identifier, string $message) {
 		$client = $this->clientService->newClient();
-		$response = $client->post(
-			$this->config->getUrl(),
-			[
-				'body' => [
-					'to' => $identifier,
-					'message' => $message,
-				],
-			]
-		);
-		$body = $response->getBody();
-		$json = json_decode($body, true);
+		// determine type of gateway
+		$response = $client->get($this->config->getUrl() . '/v1/about');
+		if ($response->getStatusCode() === 200) {
+			// New style gateway https://gitlab.com/morph027/signal-cli-dbus-rest-api
+			$response = $client->post(
+				$this->config->getUrl() . '/v1/send/' . $identifier,
+				[
+					'json' => [ 'message' => $message ],
+				]
+			);
+			$body = $response->getBody();
+			$json = json_decode($body, true);
+			if ($response->getStatusCode() !== 201 || is_null($json) || !is_array($json) || !isset($json['timestamp'])) {
+				$status = $response->getStatusCode();
+				throw new SmsTransmissionException("error reported by Signal gateway, status=$status, body=$body}");
+			}
+		} else {
+			// Try old deprecated gateway https://gitlab.com/morph027/signal-web-gateway
```

Comment:
Is this comment necessary?

Reply:
My mistake, by accident I branched the sip-gate branch off from the bugfix branch containing the signal changes which is already here: #481

I'll redo the branch from the master. The signal changes really do not belong here.